### PR TITLE
Feature/refactor hh client

### PR DESCRIPTION
**Pull request by TomerGold1998**

### CODE REVIEW

File: src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisKeysDb.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            _stringResponseParser = stringResponseParser;
+        }
+
+        /// <summary>
+        /// Get the value of a key,
+        /// If the key does not exists in the db, the default value of T will be returned
+        /// </summary>
+        /// <typeparam name="T">The expected Type of the returned value</typeparam>
+        /// <param name="key">The redis key</param>
+        /// <returns></returns>
```
Comment (VainEyal) [inline src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisKeysDb.cs]: shouldn't it be on the interface?

File: src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisDb.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Gold.Redis.HighLevelClient.Interfaces;
+using Gold.Redis.HighLevelClient.Models.Utils;
+
+namespace Gold.Redis.HighLevelClient.Db
+{
+    public class RedisDb : IRedisDb
```
Comment (VainEyal) [inline src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisDb.cs:10]: please add regions to this file

File: src/Gold.Redis/Gold.Redis.HighLevelClient/ResponseParsers/JsonResponseParser.cs [hunk 3]
```diff
@@ -24,6 +25,12 @@ public T Parse<T>(string response)
             if (response == null)
                 return default(T);
 
+            if (typeof(T) == typeof(string) || typeof(T) == typeof(String))
```
Comment (VainEyal) [inline src/Gold.Redis/Gold.Redis.HighLevelClient/ResponseParsers/JsonResponseParser.cs]: Are you sure that both checks are necessary?
maybe 'is' could help here?

File: src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisKeysDb.cs (hunk 1 above)
Comment (TomerGold1998) [reply src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisKeysDb.cs]: Yes

File: src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisDb.cs (hunk 2 above)
Comment (TomerGold1998) [reply src/Gold.Redis/Gold.Redis.HighLevelClient/Db/RedisDb.cs:10]: Added

File: src/Gold.Redis/Gold.Redis.HighLevelClient/ResponseParsers/JsonResponseParser.cs (hunk 3 above)
Comment (TomerGold1998) [reply src/Gold.Redis/Gold.Redis.HighLevelClient/ResponseParsers/JsonResponseParser.cs]: You are rigth(no need for both string and String check), but we cannot use is operator

**Review by VainEyal — APPROVED** [review]

**Comment by VainEyal** [comment]
don't forget to delete the source branch after the merge

**Pull request merged**